strings/base_collections_base.h

```diff
@@ -415,8 +414,14 @@ WINRT_EXPORT namespace winrt
 
         void Remove(K const& key)
         {
+            auto& container = static_cast<D&>(*this).get_container();
+            auto found = container.find(static_cast<D const&>(*this).wrap_value(key));
+            if (found == container.end())
+            {
+                throw hresult_out_of_bounds();
```

Comment:
Is there a reason why it's important that Remove throws when the key is not found, beyond consistency with other implementations?

Reply:
It prevents `TryRemove` from working correctly.

Reply:
Yes, but it sounds like a circular argument. `TryRemove` needs `Remove` to change. `TryRemove` is needed because `Remove` has changed. 😉

Reply:
TryRemove is needed because if the map came from somebody other than C++/WinRT, **it will throw**.

Reply:
Got it - thanks.